### PR TITLE
Add project type to support hybrids SDK projects

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
Add project type to support hybrids SDK projects 

**Motivation**
Closes: #879

### CODE REVIEW

File: cmd/init_project.go [hunk 1]
```diff
@@ -108,6 +108,7 @@ func (o *projectOptions) bindCmdlineFlags(cmd *cobra.Command) {
 		"defaults to the go package of the current working directory.")
 	cmd.Flags().StringVar(&o.project.Domain, "domain", "my.domain", "domain for groups")
 	cmd.Flags().StringVar(&o.project.Version, "project-version", project.Version2, "project version")
+	cmd.Flags().StringVar(&o.project.ProjectType, "type", project.Go, "project type")
```
[the remark this answers is not in the document]
Comment (camilamacedo86) [reply cmd/init_project.go]: Done 👍

**Comment by camilamacedo86** [comment]
> /hold
> I'm not sure if this is the right way to approach this, since using this approach plugins will be in-tree. That is not the goal.
> 

I think that would make sense we have the info in the PROJECT file which defines the spec/attributes of it. So, based on the PROJECT file we could do any further implementation. Also, in POV it could make the things simple as for example, allow us to scaffold the Makefile according to the type and add the plugins and etc, allow the plugins know how to dealing with the code, allow the tool know how to dealing with the project and etc .. WDYT?  


cc: @droot @DirectXMan12 @mengqiy

**Comment by DirectXMan12** [comment]
cc @joelanford @justinsb 

do we currently not store initial plugin info?  Agree with camilla here -- it seems like we should store the general project structure for later so that you don't have to specify `--plugin` every time.

**Comment by camilamacedo86** [comment]
Hi @mengqiy and @DirectXMan12,

Could we move forward so with this one? Would you like to change anything in the way that it was done?

**Comment by camilamacedo86** [comment]
@mengqiy it is rebased with the master now in order to pass in the prow tests.

**Comment by camilamacedo86** [comment]
HI @mengqiy,

@DirectXMan12 agreed with this change as well here; https://github.com/kubernetes-sigs/kubebuilder/pull/1171#issuecomment-553992764. WDYT, could we move forward with this one?

**Comment by joelanford** [comment]
@camilamacedo86 @DirectXMan12 @hasbro17

We are working on a proposal that will address this and should have something posted soon. In the meantime, the working POC is at https://github.com/joelanford/kubebuilder-exp if you want an early look.

The idea is:
1. Introduce kubebuilder CLI plugin interface that the CLI uses to decide how to scaffold projects, APIs, and webhooks.
2. Each plugin defines its project version. Project versions must be unique in the set of plugins registered in a binary, and ideally they would be globally unique to make different binaries more interchangeable. (i.e. "2" always means kubebuilder's Go version 2 scaffold).
3. The existing Go scaffolders built into kubebuilder will continue to be versions "1" and "2".
4. Other plugins can define other versions. For example, operator-sdk's helm and ansible plugins could use "helm:1" and "ansible:1"

I'm not opposed to introducing a new ProjectType field if that makes more sense. That probably needs more discussion.

I'm in favor of leaving this on hold, pending our proposal.

**Comment by DirectXMan12** [comment]
closing as per the feature branch and related discussion.

**Pull request closed without merging**